File: q8.cpp
class Solution
{
public:
    int myAtoi(string s)
    {
        int cw = 0;
        int i = 0;
        long long int ans = -1;
        bool flag = 0;
        while (s[i] == ' ')
        {
            i++;
        }
        s.erase(s.begin(), s.begin() + i);
        i = 0;
        if (s[i] == '-')
        {
            flag = 1;
            i++;
        }
        else if (s[i] == '+')
        {
            i++;
        }
        for (; i < s.length(); i++)
        {
            int c = s[i] - '0';
            if (c >= 0 && c <= 9)
            {
                if (ans != -1)
                {

                    ans *= 10;
                    if (ans >= INT_MAX && flag == 0)
                        return INT_MAX;
                    if (ans >= INT_MAX && flag == 1)
                        return INT_MIN;
                    ans += c;
                }
                else
                {
                    ans = c;
                }
            }
            else
            {
                if (ans == -1)
                {
                    return 0;
                }
                break;
            }
        }

        ans = ans == -1 ? 0 : ans;
        ans = flag ? -ans : ans;
        ans = ans >= INT_MAX ? INT_MAX : ans;
        ans = ans <= INT_MIN ? INT_MIN : ans;
        return ans;
    }
};

File: q8.py
class Solution:
    def myAtoi(self, s: str) -> int:
        s = s.strip()
        if len(s) == 0:
            return 0
        flag = 0
        st = 0
        res = ""

        if s[st] == '-':
            if len(s) == 1:
                return 0
            flag = 1
            st += 1
        elif s[st] == '+':
            if len(s) == 1:
                return 0
            st += 1
        if not s[st].isdigit():
            return 0
        for c in s[st:]:
            if not c.isdigit():
                break
            res += c
        ans = 0
        res = res if not flag else '-'+res
        ans = int(res)
        if ans > pow(2, 31)-1:
            ans = pow(2, 31)-1
        elif ans < -pow(2, 31):
            ans = -pow(2, 31)
        return ans
